### DISCUSSION TITLE
How can I get total count of Workflows associated with a project?

### DISCUSSION THREAD

**Discussion by TheCthulhuKid** [flyteorg/flyte#1115] [Q&A] [answered]
I can't find an easy way to get the total with Flytekit. Am I missing something?

**Comment by EngHabu** [comment]
Hi @gileadslostson,

Thank you for posting your question. Do you mind clarifying what do you mean by "get the total"? maybe a code snippet of what you are trying to do would help...

**Reply by kumare3** [reply]
@EngHabu @gileadslostson wants to know the count of workflows associated with the project. 

@gileadslostson is the usecase for some programmatic display? 

There are multiple ways
- use the ListWorkflowIds call as described [here](https://docs.flyte.org/projects/flyteidl/en/latest/protos/docs/service/service.html#adminservice)

- another option is to use flytectl as described [here](https://docs.flyte.org/projects/flytectl/en/latest/gen/flytectl_get_workflow.html#flytectl-get-workflow)

- A programmatic option to do this is to use the Python friendly Client # [list_workflow_ids api](https://github.com/flyteorg/flytekit/blob/master/flytekit/clients/friendly.py#L197)

**Comment by kumare3** [comment, marked as answer]
@EngHabu @gileadslostson wants to know the count of workflows associated with the project. 

@gileadslostson is the usecase for some programmatic display? 

There are multiple ways
- use the ListWorkflowIds call as described [here](https://docs.flyte.org/projects/flyteidl/en/latest/protos/docs/service/service.html#adminservice)

- another option is to use flytectl as described [here](https://docs.flyte.org/projects/flytectl/en/latest/gen/flytectl_get_workflow.html#flytectl-get-workflow)

- A programmatic option to do this is to use the Python friendly Client # [list_workflow_ids api](https://github.com/flyteorg/flytekit/blob/master/flytekit/clients/friendly.py#L197)

**Reply by TheCthulhuKid** [reply]
This does not return the count.

**Reply by TheCthulhuKid** [reply]
It doesn't make sense to retrieve all IDs every time I make a pagination call.

**Reply by kumare3** [reply]
ya there is not API today in flyteadmin to count the API's.

**Comment by TheCthulhuKid** [comment]
Thanks for answering. What I mean is the total count of workflows.

Pagination doesn't return the total number rather just the token for the
next set of results.

**Comment by TheCthulhuKid** [comment]
And, sorry, yes it is programmatically via the python API.

**Comment by kumare3** [comment]
@gileadslostson today there is no API that returns just the count. You will have to retrieve all ID's (these are only the unique ids in a project) and then perform a client side count. Another option is to directly go to the database.

I would love to understand the usecase, and to see if this actually make sense as a feature?

**Comment by TheCthulhuKid** [comment]
We want to use Flyte for managing our tasks and it will replace existing
functionality.
At the moment it is useful to see the total count of various elements as
well as jump to specific pages.
What I could do is get the IDs on the first call and cache the count. Not
ideal but I will look into it.
If you need more info just give me a shout :)

**Reply by kumare3** [reply]
Is this like telemetry- if so we do export Prometheus metrics from various Flyte components https://docs.flyte.org/projects/cookbook/en/latest/auto/deployment/cluster/monitoring.html#sphx-glr-auto-deployment-cluster-monitoring-py.

Also please jump into slack and we can discuss and help more if needed. Again thank you for the support
